src/api/app/controllers/webui/project_controller.rb

```diff
@@ -1182,6 +1182,13 @@ def status_check_package(p)
     currentpack = Hash.new
     pname = p.name
 
+    currentpack['requests_from'] = Array.new
+    key = @api_obj.name + '/' + pname
+    if @submits.has_key? key
+      currentpack['requests_from'].concat(@submits[key])
+    end
+    return if @ignore_pending && currentpack['requests_from'].any?
```

Comment:
That would make a good refactoring.

There are many, many places where the existing code in the ProjectController can be improved. It would be a very good idea to refactor lots of it. But I think it is best to do refactoring seperate from bug fixing. That makes reviewing much easier.

In this case, I am just fixing a small bug, mostly by reshuffling existing code.